webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala

```diff
@@ -822,46 +824,93 @@ class GitTechniqueReader(
+                                 info.techniques(techniqueId.name) = MutMap(techniqueId.version -> pack)
+                                 info.techniquesCategory(techniqueId) = parentCategoryId
+                               }
+                             case Some(versionMap) => // check for the version
+                               versionMap.get(techniqueId.version) match {
+                                 case None    => // add that version
+                                   if (updateParentCat(info, parentCategoryId, techniqueId, descriptorFile)) {
+                                     info.techniques(techniqueId.name)(techniqueId.version) = pack
+                                     info.techniquesCategory(techniqueId) = parentCategoryId
+                                   }
+                                 case Some(v) => // error, policy package version already exists
+                                   logger.error(
+                                     "Ignoring package for policy with ID %s and root directory %s because an other policy is already defined with that id and root path %s"
+                                       .format(TechniqueId, descriptorFile.getParent, info.techniquesCategory(techniqueId).toString)
+                                   )
+                               }
+                           }
                          }
-                       }.fold(
-                         err =>
-                           err match {
+                       ).fold(
+                         ex =>
+                           ex match {
                              case e: ConstraintException =>
                                s"Ignoring technique '${filePath}}' because the descriptor file is malformed. Error message was: ${e.getMessage}}".fail
-                             case e: Throwable           => s"Error when processing technique '${filePath}}': ${e.getMessage}}".fail
+                             case e: Throwable           =>
+                               s"Error when processing technique '${filePath}}': ${e.getMessage}}".fail
                            },
-                         ok => techniquesInfo.set(info)
+                         _ => techniquesInfo.set(info)
                        )
     } yield {
       ()
     }
   }
 
+  // techniqueRelativePath is the path of the technique relative to /techniques/, not to git repo root.
+  def checkTechniqueIdMatchesYaml(techniqueRelativePath: File, techniqueId: TechniqueId): Task[Unit] = {
+    import zio._
+    import zio.json.yaml._
+    import com.normation.rudder.ncf.yaml.{Technique => YamlTechnique}
+    import com.normation.rudder.ncf.yaml.YamlTechniqueSerializer._
+
+    // looking at descriptor on file path, because it's what TechniqueCompiler will look at
+    val descriptor = repo.rootDirectory / (relativePathToGitRepos.getOrElse(
+      ""
+    ) + "/" + techniqueRelativePath.getPath + "/" + TechniqueFiles.yaml)
```

Comment:
why not using better files "/" here ? 

repo.rootDirectory / (relativePathToGitRepos.getOrElse("" ) / techniqueRelativePath.getPath  / TechniqueFiles.yaml ?